### ISSUE TITLE
[8.0] timezone error on orders import

### ISSUE THREAD

**Issue by haroldtamo** [OCA/connector-magento#181]
When importing orders, the connector set the import_start_time with datetime.now() : https://github.com/OCA/connector-magento/blob/9.0/magentoerpconnect/magento_model.py#L486

the problem is datetime.now return the date in UTC when the creation date of orders is in local time zone of server (CET for mine)

So, the connector try to import orders 1 hour before their creation.

For example, I make an order in magento at 02/12/2015 7:09:53 PM and the connector log give me that : 

search for magento saleorders {'created_at': {'to': '2015/12/02 18:09:58', 'from': '2015/12/02 18:09:01'}, 'store_id': {'in': [u'1']}, 'state': {'neq': 'canceled'}} returned []

I made this change : 
import pytz
import_start_time = datetime.now(pytz.timezone('CET'))

and my orders are imported well but i'm feeling this is not a good enough solution.

How can I definitely and properly solve this ?

Thanks

**Comment by guewen** [comment]
When you say:

> when the creation date of orders is in local time zone of server (CET for mine)

Do you mean the creation date of orders on Magento?

**Comment by haroldtamo** [comment]
Yes. And the time zone of odoo server is also CET

**Comment by guewen** [comment]
Odoo's internal timezone is always UTC even if your server is setup as CET.
I'm not a Magento expert, but I think that you can setup Magento to be based internally on a UTC timezone (it should be according to http://inchoo.net/magento/guide-through-magentos-timezones/) with datetimes shown as CET  for your users. If this is achievable with Magento, I think it is the proper correction.

**Comment by haroldtamo** [comment]
Thanks for your answer.
I tried what is told here : http://inchoo.net/magento/guide-through-magentos-timezones/) on magento but when i change the timezone, it also change the date on order in the backend.
For now i kept my change.

**Comment by haroldtamo** [comment]
despite my change the import of certain orders failed aigain. 
I edit the IMPORT_DELTA_BUFFER variable (https://github.com/OCA/connector-magento/blob/9.0/magentoerpconnect/magento_model.py#L41) and set it to 600 (10 minutes) because I had set the import orders cron to every 1 minutes. So, sometimes, the cron start and meanwhile, an order is created on magento but the creation of the order on magento ends after the cron. Because of update of next_time on storeview with 30 seconds delta which is too short, this order will be never imported aigain if it took more than 30 second to record it in magento which is my case. I removed the timezone change.
